plugin/pkg/shoot/validator/admission.go

```diff
@@ -1368,6 +1369,11 @@ func validateKubeletConfig(fldPath *field.Path, machineTypes []gardencorev1beta1
 		if kubeletConfig.SystemReserved.Memory != nil {
 			reservedMemory.Add(*kubeletConfig.SystemReserved.Memory)
 		}
+		k8sGreaterOrEqual131, _ := semver.NewConstraint(">= 1.31")
+
+		if kubeletVersion != nil && k8sGreaterOrEqual131.Check(kubeletVersion) {
```

Comment:
Why did you put this here? I would have expected to see this in the "static validation" in https://github.com/gardener/gardener/blob/a68e11ff6d03f80f0aaff4f4c225255c1fe93ceb/pkg/apis/core/validation/shoot.go#L1788-L1880

Reply:
Oh, I've missed that file. Thanks! The validation is now moved there.